### PR TITLE
feat(mp-o4xl): show IV aggregate score for team pool matches

### PR DESCRIPTION
**Pull request by gitrgoliveira**
## Summary

- Team pool matches now display an individual-victories (IV) aggregate score (e.g. `2–1`) instead of `—` in both the new `PoolNumberedMatchRow` and the legacy `PoolMatchRow`.
- Added `teamIVScore(m)` helper in `bracket.jsx` that derives the IV aggregate from persisted `subResults`, mirroring Go canonical `engine.ComputeTeamSummary` logic: skips daihyosen sentinel (`position < 0`), awards IV based on persisted `winner` field per sub-bout, treats empty winner as hikiwake (no IV to either side).
- Individual matches are unchanged — `teamIVScore` returns `null` when no `subResults` exist, falling through to `formatIpponsScore`.

## Why

Team pool matches persist no `ipponsA`/`ipponsB` at the match level (all scoring lives in `subResults`). `formatIpponsScore([], [], …)` returns `""`, so the viewer displayed `—`. The fix derives a meaningful IV score from the data that IS persisted.

## Files changed

| File | What |
|---|---|
| `web-mobile/js/bracket.jsx` | Add `teamIVScore(m)` helper + window/export exposure |
| `web-mobile/js/viewer.jsx` | `PoolNumberedMatchRow` + `PoolMatchRow`: prefer `teamIVScore` over `formatIpponsScore`; update stale mp-o4xl comment; export `PoolNumberedMatchRow` for tests |
| `web-mobile/js/__tests__/team_iv_score.test.jsx` | 10 unit tests for `teamIVScore` (individual→null, team wins, hikiwake, daihyosen excluded, orientation, edge cases) |
| `web-mobile/js/__tests__/viewer_pools_viewer.test.jsx` | 3 render tests for `PoolNumberedMatchRow` team IV display |

## Screenshots

**Overview tab** — VSchedItem recent results showing IV scores (`1–2`, `0–2`) instead of `—`:

Overview tab IV scores

**Pools tab** — PoolNumberedMatchRow showing IV scores (`1–2`, `0–2`) alongside standings with IV column:

Pools tab IV scores

## Test plan

- [x] `make go/test` passes (lint + security scan + tests)
- [x] New/updated unit tests cover the change (13 new tests: 10 unit + 3 render)
- [x] Manual browser verification (for `web-mobile/` changes) — team pool match showing IV score instead of "—"
- [x] Screenshots added above (REQUIRED for any UI-affecting change)
- [x] No new console errors or warnings

Closes mp-o4xl

### CODE REVIEW

File: web-mobile/js/viewer.jsx [hunk 1]
```diff
@@ -2176,7 +2176,7 @@ const PoolMatchRow = React.memo(({ m, onClick }) => {
   const bWin = winnerName && winnerName === bName;
 
   const scoreStr = m.status === "completed"
-    ? window.formatIpponsScore(m.ipponsB, m.ipponsA, m.score, m.decision, m.encho, m.decidedByHantei)
+    ? (window.teamIVScore(m) || window.formatIpponsScore(m.ipponsB, m.ipponsA, m.score, m.decision, m.encho, m.decidedByHantei))
     : null;
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/viewer.jsx:2194]: Dismissed — this is the pre-existing `window.*` call convention used throughout viewer.jsx. The prior code called `window.formatIpponsScore(...)` at this same site, equally unguarded. Not a regression introduced by this PR.

File: web-mobile/js/viewer.jsx [hunk 2]
```diff
@@ -2333,11 +2333,11 @@ const PoolNumberedMatchRow = React.memo(({ m, num, onMatchClick }) => {
...
   // handles the empty-string/null → "—" fallback in one place.
+  // For team matches, teamIVScore derives the IV aggregate from subResults (mp-o4xl);
+  // for individual matches it returns null and formatIpponsScore is used instead.
   // ipponsB = Shiro (left), ipponsA = Aka (right) — same arg order as all other callers.
-  // Note: team pool matches typically don't persist ipponsA/B (only subResults is stored),
-  // so formatIpponsScore may return "" for those — see mp-o4xl for the follow-up.
   const scoreStr = m.status === "completed"
-    ? window.formatIpponsScore(m.ipponsB, m.ipponsA, m.score, m.decision, m.encho, m.decidedByHantei)
+    ? (window.teamIVScore(m) || window.formatIpponsScore(m.ipponsB, m.ipponsA, m.score, m.decision, m.encho, m.decidedByHantei))
     : null;
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/viewer.jsx:2355]: Same class as the thread at :2178 — pre-existing unguarded `window.*` pattern. Not a regression.

File: web-mobile/js/__tests__/viewer_pools_viewer.test.jsx [hunk 3]
```diff
@@ -335,3 +335,104 @@ describe('PoolsViewer league standings label (mp-mnwu)', () => {
...
+        { position: 2, winner: 'TeamB', sideA: 'P5', sideB: 'P6' },
+      ],
+    };
+
+    const tree = runtime.mount(PoolNumberedMatchRow, { m, num: 1 });
+    const text = collectText(tree);
+    // Should show "2–1" from teamIVScore, not "—" (the fallback for empty score)
+    expect(text).toContain('2–1');
+    expect(text).not.toBe(expect.stringContaining('—'));
+  });
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/__tests__/viewer_pools_viewer.test.jsx:412]: Good catch — fixed in ddeb9162. Changed to `.not.toContain('—')`. Class swept: no other occurrences of `.not.toBe(expect.stringContaining(...))` in the test suite.

File: web-mobile/js/bracket.jsx [hunk 4]
```diff
@@ -410,12 +433,24 @@ function BracketTree({ rounds, variant = 1, showDojo = true, onMatchClick, highl
...
 
+// matchScoreStr: unified score string for any completed match.
+// Tries teamIVScore first (team matches with subResults → "IV–IV"),
+// then falls back to formatIpponsScore. Callers pass pre-resolved ippons
+// arrays (which may be derived from scoreA/scoreB for bracket matches).
+// Returns "" when neither path produces a string (caller handles "—" fallback).
+function matchScoreStr(m, ipponsB, ipponsA) {
+  return teamIVScore(m)
+    || formatIpponsScore(ipponsB, ipponsA, m.score, m.decision, m.encho, m.decidedByHantei);
+}
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/bracket.jsx:444]: Dismissed — team match-level `decision` is always `"fought"` (each sub-bout carries its own decision individually). `decisionSuffix({decision: "fought"})` produces `""`, so no metadata is lost. The IV string (`2–1`) is the correct representation at the team-match level; individual-bout decisions (kiken, encho, etc.) are visible when expanding the match detail card which renders each sub-bout with its own `formatIpponsScore` + suffix.

File: web-mobile/js/__tests__/viewer_pools_viewer.test.jsx [hunk 5]
```diff
@@ -335,3 +343,108 @@ describe('PoolsViewer league standings label (mp-mnwu)', () => {
...
+        { position: 1, winner: 'TeamA', sideA: 'P3', sideB: 'P4' },
+        { position: 2, winner: 'TeamB', sideA: 'P5', sideB: 'P6' },
+      ],
+    };
+
+    const tree = runtime.mount(PoolNumberedMatchRow, { m, num: 1 });
+    const text = collectText(tree);
+    // Should show "2–1" from teamIVScore, not "—" (the fallback for empty score)
+    expect(text).toContain('2–1');
+    expect(text).not.toBe(expect.stringContaining('—'));
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/__tests__/viewer_pools_viewer.test.jsx]: Fixed in the same commit (ddeb9162) — this was the same line as thread #3, changed from `.not.toBe(expect.stringContaining('—'))` to `.not.toContain('—')`.

File: web-mobile/js/bracket.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/bracket.jsx:444]: Dismissed — the `(ipponsB, ipponsA)` parameter order mirrors `formatIpponsScore` exactly (which `matchScoreStr` delegates to), and the doc comment at :436-440 already documents orientation. Renaming here but not in `formatIpponsScore` would create an inconsistency; renaming both is out of scope for this PR.
